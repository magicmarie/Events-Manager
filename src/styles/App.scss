// landing page
h2 {
    color: #1b7999;
    text-align: center;
  }

div {
  font-family: Quantico;
} 


.inner {
  width: 94%;
  margin: 2%;
}

.card-content {
  text-align: right;
}

.card-date {
  float: left;
  text-align: left;
  font-style: italic;
}

.event-card {
  text-align: left;
  float: left;
  width: 70%;
  margin-left: 0.5%;
  margin-right: 0.5%;
}

.img-tile {
  float: right;
  width: 27%;
  margin-left: 1%;
}

.events {
  float: left;
  width: 45%;
  text-align: center;
  height: 100%;
  margin-left: 5%;
  min-width: 250px;
}

.tile {
  height: 300px;
}

.nav-links {
  color: white;
}

.nav-div {
  margin-left: 2%;
}

.event {
  width: 90%;
  float: left;
  margin-left: 5%;
  margin-top: 5px;
  margin-bottom: 5px;
  min-width: 230px;
  border-bottom-style: solid;
  border-radius: 10px;
  border-width: 1px;
  border-color: rgb(200, 200, 200);
}

a {
  text-decoration: none;
}

  @media only screen and (max-width: 600px) {
  .event {
    width: 80%;
    float: none;
    margin-left: 10%;
  }

.img-tile {
    margin: 10px 2% 10px 2%;
    width: 80%;
    float: none;
  }
}

// navbar
.search-container button {
  float: right;
  padding: 6px 10px;
  margin-top: 8px;
  margin-right: 16px;
  background: #ddd;
  font-size: 17px;
  border: none;
  cursor: pointer;
}

.search-container button:hover {
  background: #ccc;
}

@media screen and (max-width: 600px) {
  .search-container {
    float: none;
  }
  .collapse input[type=text], .collapse .search-container button {
    float: none;
    display: block;
    text-align: left;
    width: 100%;
    margin: 0;
    padding: 14px;
  }
  .collapse input[type=text] {
    border: 1px solid #ccc;  
  }
}

// register page
.register{
  margin-top: 3%;
  padding: 3%;
}
.register-left{
  text-align: center;
  margin-top: 2%;
  background: #f8f9fa;
}
.paragraph {
  text-align: center;
}
.register-left input{
  border: none;
  border-radius: 1.5rem;
  padding: 2%;
  width: 90%;
  background: #f8f9fa;
  font-weight: bold;
  color: #383d41;
  margin-top: 30%;
  margin-bottom: 3%;
  cursor: pointer;
}
.register-right{
  background: #f8f9fa;
  margin-top: 2%;
}

.register-left img{
  margin-top: 15%;
  margin-bottom: 5%;
  width: 25%;
  -webkit-animation: mover 2s infinite  alternate;
  animation: mover 1s infinite  alternate;
}
@-webkit-keyframes mover {
  0% { transform: translateY(0); }
  100% { transform: translateY(-20px); }
}
@keyframes mover {
  0% { transform: translateY(0); }
  100% { transform: translateY(-20px); }
}
.register-left p{
  font-weight: lighter;
  padding: 12%;
  margin-top: -9%;
}
.register .register-form{
  padding: 10%;
  margin-top: 2%;
}
.btnRegister{
  border: none;
  margin-left: 25%;
  border-radius: 1.5rem;
  padding: 2%;
  background: navy;
  color: #fff;
  font-weight: 600;
  width: 50%;
  cursor: pointer;
}
.register-heading{
  text-align: center;
  color: #495057;
}
.reminder {
  margin-top: 4%;
  margin-left: 17%;
}

// social login 
.social-btn {
  margin-top: 15%;
}
.form-control, .btn {
    min-height: 38px;
    border-radius: 2px;
}
.login-btn {        
    font-size: 15px;
    font-weight: bold;
}
.or-seperator {
    margin: 20px 0 10px;
    text-align: center;
    border-top: 1px solid #ccc;
    background: #f8f9fa;
}
.social-btn .btn {
    margin: 34px 0 0 20px;
    font-size: 18px;
    text-align: left; 
    line-height: 40px;       
}
.social-btn .btn i {
  float: left;
  margin: 4px 15px  0 5px;
      min-width: 15px;
}
.input-group-addon .fa{
  font-size: 18px;
}

.margin-set {
  margin-top: 3%;
}
