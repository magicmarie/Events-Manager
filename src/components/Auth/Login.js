import React from "react";
import { Link } from "react-router-dom";
import "../../styles/App.scss";

const Login = ({
  email,
  password,
  handleOnChange,
  handleLogin
}) => {
  return (
    <div className="container register">
      <div className="row">
        <div className="col-md-12">
          <h3 className="register-heading">Welcome back to Events Manager!</h3>
          <p className="paragraph">
            A social that creates a community of like minded planners to foster
            inspiration and
            <br />
            innovation by leveraging the modern web.
          </p>
        </div>
        <div className="col-md-4 register-left">
          <div className="text-center social-btn">
            <a href="#" className="btn btn-primary btn-block">
              <i className="fa fa-facebook" /> Sign in with <b>Facebook</b>
            </a>
            <a href="#" className="btn btn-info btn-block">
              <i className="fa fa-twitter" /> Sign in with <b>Twitter</b>
            </a>
            <a href="#" className="btn btn-danger btn-block">
              <i className="fa fa-google" /> Sign in with <b>Google</b>
            </a>
          </div>
        </div>
        <div className="or-separator" />
        <div className="col-md-7 register-right">
          <div className="tab-content" id="myTabContent">
            <div
              className="tab-pane fade show active"
              id="home"
              role="tabpanel"
              aria-labelledby="home-tab"
            >
              <form className="row register-form" onSubmit={handleLogin}>
                <div className="col-md-10">
                  <div className="form-group">
                    <input
                      type="email"
                      name="email"
                      onChange={handleOnChange}
                      className="form-control"
                      placeholder="Your Email *"
                    />
                  </div>
                  <div className="form-group">
                    <input
                      type="password"
                      name="password"
                      onChange={handleOnChange}
                      className="form-control"
                      placeholder="Password *"
                    />
                  </div>
                  <input
                    type="submit"
                    className="btnRegister"
                    value="Login"
                  />
                </div>
                <div className="reminder">
                  {" "}
                  Don't have an Account yet?, <Link to="/signup"> Signup</Link>{" "}
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Login;
