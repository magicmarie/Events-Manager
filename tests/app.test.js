import React from "react";
import { shallow } from "enzyme";
import App from "../src/containers/index";
import Login from "../src/components/Auth/Login";
import Home from "../src/components/home";
import events from "./mock_data/moxios_mock";
// the package for mocking/faking the Redux store
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

// middlewares, empty array for now
const middlewares = [];
const mockStore = configureStore(middlewares);
const initialState = {};
//the mocked store to pass to the Provider's store prop
const mockedStore = mockStore(initialState);

const props = {
  events: events,
  getEventsAction: jest.fn()
};

describe("App", () => {
  it("should render without crashing", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.instance()).toMatchSnapshot();
  });

  it("renders the login component", () => {
    const wrapper = shallow(<Login />);
    expect(wrapper.instance()).toMatchSnapshot();
  });

  it("render the home page", () => {
    const wrapper = shallow(
      <Provider store={mockedStore}>
        <Home {...props} />
      </Provider>
    );
    expect(wrapper.instance()).toMatchSnapshot();
  });
});
