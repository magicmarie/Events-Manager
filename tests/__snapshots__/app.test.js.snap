// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<div>
  <h2>
    Welcome to Events Manager 
  </h2>
</div>
`;
